akka-docs-dev/rst/scala/stream-customize.rst

```diff
@@ -217,239 +371,55 @@ decision.
-
-.. warning::
-   There is one very important rule to remember when working with a ``Stage``. **Exactly one** method should be called
-   on the **currently passed** :class:`Context` **exactly once** and as the **last statement of the handler** where the return type
-   of the called method **matches the expected return type of the handler**. Any violation of this rule will
-   almost certainly result in unspecified behavior (in other words, it will break in spectacular ways). Exceptions
-   to this rule are the query methods ``isHolding()`` and ``isFinishing()``
-
-To illustrate these concepts we create a small :class:`PushPullStage` that implements the ``map`` transformation.
-
-|
-
-.. image:: ../images/stage_map.png
-   :align: center
-   :width: 300
-
-|
-
-Map calls ``ctx.push()`` from the ``onPush()`` handler and it also calls ``ctx.pull()`` form the ``onPull``
-handler resulting in the conceptual wiring above, and fully expressed in code below:
-
-.. includecode:: code/docs/stream/FlowStagesSpec.scala#one-to-one
-
-Map is a typical example of a one-to-one transformation of a stream. To demonstrate a many-to-one stage we will implement
-filter. The conceptual wiring of ``Filter`` looks like this:
-
-|
-
-.. image:: ../images/stage_filter.png
-   :align: center
-   :width: 300
-
-|
-
-As we see above, if the given predicate matches the current element we are propagating it downwards, otherwise
-we return the "ball" to our upstream so that we get the new element. This is achieved by modifying the map
-example by adding a conditional in the ``onPush`` handler and decide between a ``ctx.pull()`` or ``ctx.push()`` call
-(and of course not having a mapping ``f`` function).
-
-.. includecode:: code/docs/stream/FlowStagesSpec.scala#many-to-one
-
-To complete the picture we define a one-to-many transformation as the next step. We chose a straightforward example stage
-that emits every upstream element twice downstream. The conceptual wiring of this stage looks like this:
-
-|
-
-.. image:: ../images/stage_doubler.png
-   :align: center
-   :width: 300
-
-|
-
-This is a stage that has state: the last element it has seen, and a flag ``oneLeft`` that indicates if we
-have duplicated this last element already or not. Looking at the code below, the reader might notice that our ``onPull``
-method is more complex than it is demonstrated by the figure above. The reason for this is completion handling, which we
-will explain a little bit later. For now it is enough to look at the ``if(!ctx.isFinishing)`` block which
-corresponds to the logic we expect by looking at the conceptual picture.
-
-.. includecode:: code/docs/stream/FlowStagesSpec.scala#one-to-many
-
-Finally, to demonstrate all of the stages above, we put them together into a processing chain, which conceptually
-would correspond to the following structure:
 
-|
 
-.. image:: ../images/stage_chain.png
-   :align: center
-   :width: 650
+Detached graph stages
```

Comment:
I wouldn't call them "detached" anymore. This is a deeply technical matter when a stage can be considered "detached" that we use internally. Back then, DetachedStage expressed *exactly* that concept, but GraphStage is no longer a clear example of that. I.e. you might want to write "rate-decoupled" since that is the most common use-case for users.